stages/org.osbuild.zipl-secex.inst

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/python3
+"""
+Install the Z Initial Program Loader (zipl)
+
+This will install the Z Initial Program Loader for the
+specified `kernel` with "IBM Secure Execution" enabled.
+The other required parameter are
+- `location`, which specifies is the offset of the partition containing `/se`
+within the device that is specified via `disk`.
+- `ignition_gpg_pubkey`, which specifies filename for public GPG key
+- `hostkey`, which specifies Hostkey for given IBM Z machine
+- `build_id`, which specifies CoreOS build
+"""
+
+import libarchive
+import os
+import subprocess
+import sys
+import tempfile
+
+import osbuild.api
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "required": ["kernel", "location", "ignition_gpg_pubkey", "hostkey", "build_id"],
+  "properties": {
+    "kernel": {
+      "type": "string"
+    },
+    "location": {
+      "type": "number"
+    },
+    "hostkey": {
+      "type": "string"
+    },
+    "ignition_gpg_pubkey": {
+       "type": "string"
+    },
+    "build_id": {
+       "type": "string"
+    },
+    "kernel_opts_append": {
+      "description": "Additional kernel options to add to the discovered options",
+      "type": "array",
+      "items": {
+        "description": "A single kernel command line option",
+        "type": "string"
+      }
+    },
+    "blocksize": {
+      "type": "number",
+      "default": 512
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true,
+  "required": ["disk"],
+  "properties": {
+    "disk": {
+      "type": "object",
+      "additionalProperties": true
+    }
+  }
+},
+"mounts": {
+  "type": "array",
+  "minItems": 1
+}
+"""
+
+def parse_blsfile(blsfile):
+    params = {}
+    with open(blsfile, "r", encoding="utf8") as bls:
+        for line in bls:
+            key, value = line.split(' ', 1)
+            params[key] = value.strip()
+    return params
+
+
+def find_kernel(root, kernel: str):
+    base = f"{root}/boot/loader/entries"
+    for dirent in os.scandir(base):
+        fn, ext = os.path.splitext(dirent.name)
+        if ext != ".conf" or fn.endswith("rescue"):
+            continue
+        blsfile = f"{base}/{dirent.name}"
+        params = parse_blsfile(blsfile)
+
+        version = params["version"]
+        if version != kernel:
+            continue
+
+        linux = root + params["linux"]
+        initrd = root + params["initrd"]
+        options = params.get("options", "")
+        return linux, initrd, options
+
+    raise ValueError(f"Unknown kernel: {kernel}")
```

Comment:
Maybe we should host these into a shared library since it's copy/pasted from `org.osbuild.zipl.inst`?

Reply:
i.e. see what @ravanelli is doing in https://github.com/osbuild/osbuild/pull/1615/files